### PR TITLE
Init

### PR DESCRIPTION
**Pull request by DeepDiver1975**

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -0,0 +1,104 @@
+workspace:
+  base: /drone
+  path: src
+
+pipeline:
+  restore:
+    image: plugins/s3-cache:1
+    pull: true
+    secrets: [ cache_s3_server, cache_s3_access_key, cache_s3_secret_key ]
+    pull: true
```
Comment (tboerger) [inline .drone.yml]: Duplicated line

File: .drone.yml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    restore: true
+    when:
+      local: false
+
+  composer:
+    image: webhippie/php-apache:latest
```
Comment (tboerger) [inline .drone.yml]: `owncloudci/php:7.1`

File: .drone.yml [hunk 3]
```diff
@@ -0,0 +1,104 @@
+workspace:
+  base: /drone
+  path: src
```
Comment (tboerger) [inline .drone.yml]: set the `path` to `server/apps/files_primary_s3`, than you can sync the core to `/drone/server` and you got everything in place.

Comment (DeepDiver1975) [reply .drone.yml]: I doubt that I can call git clone into an existing folder? or am I missing anything? THX

Comment (tboerger) [reply .drone.yml]: true, that have to happen before. Or requires something like rsync/cp -rf

Comment (DeepDiver1975) [reply .drone.yml]: or can we customize the clone step?

Comment (tboerger) [reply .drone.yml]: It's possible to clone something else first, this is just written out of my head so it needs to be confirmed that it works as intended.

```
workspace:
  base: /drone
  path: server/apps/files_primary_s3

clone:
  core:
    image: plugins/git
    remote: https://github.com/owncloud/core.git
    path: /drone/server
    ref: refs/heads/master
    recursive: true

  git:
    image: plugins/git

pipeline:
  restore:
    image: plugins/s3-cache:1
    pull: true
    secrets: [ cache_s3_server, cache_s3_access_key, cache_s3_secret_key ]
    restore: true
    when:
      local: false
```

This changed `clone` step SHOULD clone the core into `/drone/server` and afterwards the app into `/drone/server/apps/files_primary_s3`. Attributes like the `ref` can be replaced by matrix variables later on to run the tests against different core versions.

**Pull request merged**